### PR TITLE
test: Cypress - Added Tags to test cases - @tag.GSheet, @tag.Datasource,  @tag.JS, @tag.IDE, @tag.Settings, @tag.Binding

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
Run GSheet test cases -    `/ok-to-test tags="@tag.GSheet"`
Run JS test cases   -  `  /ok-to-test tags="@tag.JS"`
Run IDE test cases    -  `  /ok-to-test tags="@tag.IDE"`
Run Binding test cases - `  /ok-to-test tags="@tag.Binding"`
Few datasource testcases (tags are added to few) using         -    ` /ok-to-test tags="@tag.Datasource"`
To run both GSheet and Datasource tagged testcases -` /ok-to-test tags="@tag.Datasource,@tag.GSheet"`
To run both Binding and IDE tagged testcases - ` /ok-to-test tags="@tag.Binding,@tag.IDE"`
To run both Binding, IDE, JS tagged testcases - ` /ok-to-test tags="@tag.Binding,@tag.IDE,@tag.JS"`

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Enhanced test suite categorization with the addition of metadata tags for improved filtering and organization.

- **Refactor**
  - Reorganized test suite structures for clarity and maintainability.
  - Updated test descriptions and assertions for better test coverage.

- **Bug Fixes**
  - Modified test cases to include additional validation steps for bug fix verification.

- **Documentation**
  - Updated test suite descriptions with new tags to reflect changes in testing strategies.

- **Chores**
  - Adjusted file paths for test specifications to align with the updated testing focus.

### CODE REVIEW

File: app/client/cypress/e2e/GSheet/WidgetBinding_AllAccess_Spec.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const workspaceName = "gsheet apps";
 const dataSourceName = "gsheet";
 let appName = "gsheet-app";
 let spreadSheetName = "test-sheet";
-describe("GSheet-widget binding", { tags: ["@tag.Datasource"] }, function () {
+describe("GSheet-widget binding", { tags: ["@tag.Datasource", "@tag.Datasource", "@tag.GSheet"] }, function () {
```
[the remark this answers is not in the document]
Comment (sharanya-appsmith) [reply app/client/cypress/e2e/GSheet/WidgetBinding_AllAccess_Spec.ts:24]: Done

File: app/client/cypress/limited-tests.txt [hunk 2]
```diff
@@ -1,6 +1,5 @@
 # To run only limited tests - give the spec names in below format:
-cypress/e2e/Regression/ClientSide/Templates/Fork_Template_spec.js
-
+cypress/e2e/Regression/Apps/CurrencyInputIssue_Spec.js
```
Comment (mohanarpit) [inline app/client/cypress/limited-tests.txt:2]: @sharanya-appsmith Will this create any impact to the rest of the workflow runs?

Comment (sharanya-appsmith) [reply app/client/cypress/limited-tests.txt:2]: When people try to push changes this will create a conflict. Will revert this.

Comment (sharanya-appsmith) [reply app/client/cypress/limited-tests.txt]: Done

**Review by mohanarpit — APPROVED** [review]

**Comment by sharanya-appsmith** [comment]
@mohanarpit  can you please merge this? ci-test-result not updated and couldn't merge.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**